lib/utils/objectQueryTracker.ts

```diff
@@ -0,0 +1,32 @@
+import { LocalStorage } from "./localStorage";
+
+interface ObjectQueryMap {
+  [objectId: string]: [queryId: string, timestamp: number];
+}
+
+const PREFIX = "AlgoliaObjectQueryCache_";
+
+function getCache(index: string): ObjectQueryMap {
+  return LocalStorage.get(`${PREFIX}${index}`) ?? {};
+}
+function setCache(index: string, objectQueryMap: ObjectQueryMap): void {
+  LocalStorage.set(`${PREFIX}${index}`, objectQueryMap);
+}
+
+export function storeQueryForObject(
+  index: string,
+  objectID: string,
+  queryID: string
+): void {
+  const objectQueryMap = getCache(index);
+
+  objectQueryMap[objectID] = [queryID, Date.now()];
```

Comment:
Sorry if obvious, but what's the reason we're not using the event timestamp?

Is it worth checking for existing stored query IDs and comparing timestamps before setting a new queryID-objectID association? E.g., is it possible that a customer might pass a `timestamp` for a click event that came _before_ the timestamp stored in local storage? I assume we'd want the latest one.

Reply:
It's theoretically _possible_ that someone might fire a click on the frontend with an older `timestamp` but I think that would be pretty unlikely. Anyone passing a timestamp to a search-insights event is most likely going to be using `Date.now()` themselves.

Really I'm just adding a timestamp here so that I know when to nuke it. The last `queryID` to be set always overwrites the existing one.

Or am I missing a pretty common usecase?